### PR TITLE
Add Entity structure definition

### PR DESCRIPTION
**Pull request by perdasilva**
This PR adds the `Entity` definition as a structure with an `Id` and `Properties` (string->string).
It also adds ginkgo and gomega to the project and uses them for the unit tests.

### CODE REVIEW

**Review by timflannagan — APPROVED** [review]
lgtm once the linting job passes

File: internal/source/entity.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+func NewEntity(id EntityID, properties map[string]string) *Entity {
+	return &Entity{
+		id:         id,
+		properties: properties,
+	}
+}
+
+func (e *Entity) ID() EntityID {
+	return e.id
+}
```
Comment (awgreene) [inline internal/source/entity.go:27]: Was this method created to address the concern that the ID could be modified after creation? Should we explicitly call out that the field is immutable as a comment on the structure?

**Review by awgreene — APPROVED** [review]
Left a nit, but this looks good otherwise.
/lgtm

File: internal/source/entity.go (hunk 1 above)
Comment (timflannagan) [reply internal/source/entity.go:27]: The underlying `id` field in the Entity structure is unexported, so it should be immutable after creation of this type given there's no setter-esq method available. This method is just exposing a way to get the value of the underlying id field.

File: internal/source/entity_test.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package source_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/operator-framework/deppy/internal/source"
```
Comment (timflannagan) [inline internal/source/entity_test.go:6]: The linter is complaining about this grouping. It wants the internal package to be declared separate from the third-party packages (e.g. a newline between line 5 and line 6).

File: internal/source/entity.go (hunk 1 above)
Comment (perdasilva) [reply internal/source/entity.go:27]: Right, I was just protecting the data of the entity from mutation

**Pull request merged**